tasks/agent.py

```diff
@@ -640,16 +641,21 @@ def clean(ctx):
 
 
 @task
-def version(ctx, url_safe=False, git_sha_length=7, major_version='7'):
+def version(ctx, url_safe=False, omnibus_format=False, git_sha_length=7, major_version='7'):
     """
     Get the agent version.
     url_safe: get the version that is able to be addressed as a url
+    omnibus_format: performs the same transformations omnibus does on version names to
+                    get the exact same string that's used in package names
     git_sha_length: different versions of git have a different short sha length,
                     use this to explicitly set the version
                     (the windows builder and the default ubuntu version have such an incompatibility)
     """
-    print(
-        get_version(
-            ctx, include_git=True, url_safe=url_safe, git_sha_length=git_sha_length, major_version=major_version
-        )
+    version = get_version(
+        ctx, include_git=True, url_safe=url_safe, git_sha_length=git_sha_length, major_version=major_version
     )
+    if omnibus_format:
+        # See: https://github.com/DataDog/omnibus-ruby/blob/datadog-5.5.0/lib/omnibus/packagers/deb.rb#L599
+        version = re.sub('-', '~', version)
+        version = re.sub(r'[^a-zA-Z0-9\.\+\:\~]+', '_', version)
+    print(version)
```

Comment:
if this replicates the "safe" version format of omnibus for deb packages, should the format be different for RPMs or Windows MSIs? (and then be passed with different env vars?)

Reply:
Good question. In theory yes, we'd need to have one format for each package type (deb, rpm, msi, pkg). However, there are a few things that allow us in practice to have only one trigger for everything:
- the deb and rpm safe version formats are identical (the only difference is an additional rule on Wind River Linux: https://github.com/DataDog/omnibus-ruby/blob/03ed9e254cf2913c51dccea303718b3a1b63f59e/lib/omnibus/packagers/rpm.rb#L717-L725, which doesn't apply to us). Moreover, of the two rules, we actually really only use the first one (because we always use `inv agent.verssion --url-safe`).
- the msi version name uses the raw version. The only difference with the deb / rpm versions is therefore that dashes were replaced by tildes. We're already doing the reverse operation in agent-release-management to get the correct msi name.
- the pkg version name uses the raw version + a variation of the second rule (where a dash is used in place of an underscore: https://github.com/DataDog/omnibus-ruby/blob/03ed9e254cf2913c51dccea303718b3a1b63f59e/lib/omnibus/packagers/pkg.rb#L307-L322). Once again, replacing tildes by dashes (+ replacing underscore by dashes if we ever end up using the second rule for some reason) in agent-release-management is enough. We're already replacing tildes by dashes in agent-release-management.

But passing directly the correct version specific to each package is a future improvement we can consider doing indeed (the benefit being that we're 100% sure that we're using the correct version name for each package, not relying on the above facts, the drawback being that manual testing / triggering of agent-release-management pipelines is going to be more tiresome).

I'll add a comment explaining the above to the PR.